### PR TITLE
Remove a lot of usages of legacy globals

### PR DESCRIPTION
**Pull request by Nokel81**

### CODE REVIEW

File: src/renderer/components/error-boundary/error-boundary.tsx [hunk 1]
```diff
@@ -32,7 +38,7 @@ export class ErrorBoundary extends React.Component<ErrorBoundaryProps, State> {
 
   back = () => {
     this.setState({ error: undefined, errorInfo: undefined });
-    navigation.goBack();
+    this.props.observableHistory.goBack();
```
Comment (Iku-turso) [inline src/renderer/components/error-boundary/error-boundary.tsx:41]: Heyy, check out [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) for some cool insight ;)

File: .eslintrc.js [hunk 2]
```diff
@@ -283,6 +283,15 @@ module.exports = {
...
             },
+            {
+              "group": [
+                "**/extensions/as-legacy-globals-for-extension-api/as-legacy-global-function-for-extension-api",
+                "**/extensions/as-legacy-globals-for-extension-api/as-legacy-global-object-for-extension-api-with-modifications",
+                "**/extensions/as-legacy-globals-for-extension-api/as-legacy-global-object-for-extension-api",
+                "**/extensions/as-legacy-globals-for-extension-api/as-legacy-global-singleton-object-for-extension-api",
+              ],
+              message: "No importing the legacy global functions in non-ExtensionApi code",
+            },
```
Comment (Iku-turso) [inline .eslintrc.js:294]: Love this.

**Review by jansav — COMMENTED** [review]
Cool stuff. Will approve when CI passes.

**Review by Iku-turso — COMMENTED** [review]
Seems like Christmas came early with all this annihilation of legacy :) Would approve, but there seems to be some error in CI.

**Pull request merged**